### PR TITLE
add example_for_toolbar xml and java as example of how to integrate a…

### PR DESCRIPTION
**Pull request by DonnA98**
… dynamic toolbar in activities

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,9 +35,14 @@ android {
     }
     lintOptions {
         disable 'InvalidPackage' // Toothpick runtime includes references to javax.inject which have
-                                 // runtime retention even if they're not used since we use registries
-                                 // for injection instead of reflection.
+        // runtime retention even if they're not used since we use registries
+        // for injection instead of reflection.
     }
+    // added
```
Comment (roosemberth) [inline app/build.gradle]: Can you please remove these useless comments?

File: app/build.gradle [hunk 2]
```diff
@@ -35,9 +35,14 @@ android {
...
+    dataBinding {
+        enabled = true
+    }
+    // end added
```
Comment (roosemberth) [inline app/build.gradle]: Please remove

File: app/src/main/java/ch/epfl/sweng/erpa/activities/DependencyConfigurationAgnosticActivity.java [hunk 3]
```diff
@@ -20,7 +20,9 @@
 import toothpick.Toothpick;
 
 public abstract class DependencyConfigurationAgnosticActivity extends Activity {
-    @Inject @Named("Dependency Configurators") Set<DependencyConfigurator> coordinators;
+    @Inject
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/DependencyConfigurationAgnosticActivity.java]: Having a line for each annotation makes the code harded to overview and unnesessary long. In general we want to keep groups of annotations (such as normal fields, autoinjected fields, automatic constructors, lombok on the future, ...)

Please modify your editor settings not to manually wrap (except for class annotations).
Please modify the code to have all field member, methods and parameter annotations in the same line

File: app/src/main/java/ch/epfl/sweng/erpa/activities/SelectRemoteServicesProviderActivity.java [hunk 4]
```diff
@@ -47,9 +51,10 @@ protected void onCreate(Bundle savedInstanceState) {
         });
     }
 
-    @OnClick(R.id.rspSelectionSubmit) public void rspSelect() {
+    @OnClick(R.id.rspSelectionSubmit)
+    public void rspSelect() {
         @SuppressLint("FindViewByIdCast")  // Linter is wrong. This will always be a RadioButton
-        Optional<RadioButton> maybeSelection = Optional.ofNullable(
+                Optional<RadioButton> maybeSelection = Optional.ofNullable(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/SelectRemoteServicesProviderActivity.java]: Remove tab?

File: app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java [hunk 5]
```diff
@@ -26,29 +26,31 @@
...
 public class RemoteServicesProviderCoordinator implements InvocationHandler,
-                                                          DependencyConfigurator<RemoteServicesProvider> {
+        DependencyConfigurator<RemoteServicesProvider> {
+    private final String REMOTE_PROVIDER_KEY;
+    private final Context applicationContext;
     /*
      * A word or wary: This singleton may be called prior to an activity creation and may start
      * SelectRemoteServicesProviderActivity. Be conservative about injecting stuff here since it
      * may lead to an infinite dependency loop.
      */
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java]: This comment should be the first thing on the class definition, please move the lines above after the comment

File: app/src/main/java/ch/epfl/sweng/erpa/activities/SelectRemoteServicesProviderActivity.java [hunk 6]
```diff
@@ -47,10 +47,11 @@ protected void onCreate(Bundle savedInstanceState) {
         });
     }
 
-    @OnClick(R.id.rspSelectionSubmit) public void rspSelect() {
+    @OnClick(R.id.rspSelectionSubmit)
+    public void rspSelect() {
```
Comment (rykeuse) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/SelectRemoteServicesProviderActivity.java]: Normal que ce soit toujours sur deux lignes ?

File: app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java [hunk 7]
```diff
@@ -26,29 +26,28 @@
 /**
  * This is responsible of memoizing and proxying the selected RemoteServicesProvider.
  * This class' life cycle is managed by Toothpick.
- *
+ * <p>
  * Hereafter rsp = Remote Services Provider.
- *
+ * <p>
```
Comment (rykeuse) [inline app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java]: ???

Comment (roosemberth) [reply app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java]: Javadoc formatting

File: app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java [hunk 8]
```diff
@@ -26,29 +26,28 @@
...
     private final Context applicationContext;
-
+    @Inject @Named("application") Scope applicationScope;
+    @Inject SharedPreferences sharedPreferences;
     @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     private Optional<RemoteServicesProvider> currentProvider = Optional.empty();
 
-    @Inject public RemoteServicesProviderCoordinator(@Named("application") Scope scope, Context appCtx) {
+    @Inject
+    public RemoteServicesProviderCoordinator(@Named("application") Scope scope, Context appCtx) {
```
Comment (rykeuse) [inline app/src/main/java/ch/epfl/sweng/erpa/operations/RemoteServicesProviderCoordinator.java]: 2 lignes

File: app/src/main/res/layout/activity_example_for_toolbar.xml [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+
+        <!-- Inside the DrawerLayout insert a CoordinatorLayout to use scrolling effect within the Layout-->
+        <android.support.design.widget.CoordinatorLayout
+            android:id="@+id/CoordinatorLayout"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+
+            <!-- The AppBarLayout is a wrapper for the toolBar-->
+            <!-- Use it to have scrolling effect-->
+            <!-- The AppBarLayout has to be the firs child of the coordinatorLayout-->
```
Comment (rykeuse) [inline app/src/main/res/layout/activity_example_for_toolbar.xml]: typo (firs -> first)

File: app/src/main/res/layout/activity_example_for_toolbar.xml [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+                <LinearLayout
+                    android:id="@+id/LinearLayout1"
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:orientation="vertical">
+
+                    <TextView
+                        android:id="@+id/textview1"
+                        android:layout_width="fill_parent"
+                        android:layout_height="70dp"
```
Comment (rykeuse) [inline app/src/main/res/layout/activity_example_for_toolbar.xml]: Fixed size ?

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -34,4 +34,5 @@
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent" />
+
 </android.support.constraint.ConstraintLayout>
```
Comment (rykeuse) [inline app/src/main/res/layout/activity_main.xml]: newline needed

**Pull request merged**